angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -3,397 +3,80 @@ import {
-                }"
-              >
-                <a
-                  *ngIf="!narrowLink.routerLink"
-                  sprkLink
-                  variant="unstyled"
-                  class="sprk-c-MastheadAccordion__summary"
-                  [attr.href]="narrowLink.href"
-                  [analyticsString]="narrowLink.analyticsString"
-                >
-                  <span class="sprk-c-MastheadAccordion__heading">
-                    <sprk-icon
-                      [iconType]="narrowLink.leadingIcon"
-                      additionalClasses="
-                        sprk-c-Icon--filled-current-color
-                        sprk-c-Icon--stroke-current-color
-                        sprk-c-Icon--xl
-                        sprk-u-mrs
-                      "
-                      *ngIf="narrowLink.leadingIcon"
-                    ></sprk-icon>
-                    {{ narrowLink.text }}
-                  </span>
-                </a>
-                <a
-                  *ngIf="narrowLink.routerLink"
-                  sprkLink
-                  variant="unstyled"
-                  class="sprk-c-MastheadAccordion__summary"
-                  [routerLink]="narrowLink.routerLink"
-                  [analyticsString]="narrowLink.analyticsString"
-                >
-                  <span class="sprk-c-MastheadAccordion__heading">
-                    <sprk-icon
-                      [iconType]="narrowLink.leadingIcon"
-                      additionalClasses="
-                        sprk-c-Icon--filled-current-color
-                        sprk-c-Icon--stroke-current-color
-                        sprk-c-Icon--xl
-                        sprk-u-mrs
-                      "
-                      *ngIf="narrowLink.leadingIcon"
-                    ></sprk-icon>
-                    {{ narrowLink.text }}
-                  </span>
-                </a>
-              </li>
-            </div>
-          </div>
-        </sprk-masthead-accordion>
-        <ng-content select="[narrowNavFooter]"></ng-content>
-      </nav>
+        <ng-content select="[sprkMastheadBranding]"></ng-content>
+        <ng-content select="[sprkMastheadNavItem]"></ng-content>
+        <ng-content select="[sprkMastheadNavItems]"></ng-content>
+      </sprk-stack>
+      <ng-content></ng-content>
     </header>
   `,
 })
-export class SprkMastheadComponent implements AfterContentInit {
+export class SprkMastheadComponent implements AfterViewInit {
   /**
    * @ignore
    */
   constructor(private renderer: Renderer2, router: Router) {
     router.events.subscribe((event: Event) => {
-      if (event instanceof NavigationEnd) {
-        this.closeNarrowNav();
+      /**
+       * If page is changed by the router
+       * and they have a collapsibleNav
+       * we want to make sure it is closed
+       * when the new page loads if it is open.
+       */
+      if (!this.collapsibleNavDirective) {
```

Comment:
We now have an additional check for if the collapsible nav exists because users can now choose to not have one.